src/psyclone/psyir/backend/fortran.py

```diff
@@ -696,7 +714,12 @@ def gen_typedecl(self, symbol, include_visibility=True):
                 f"Fortran backend cannot generate code for symbol "
                 f"'{symbol.name}' of type '{type(symbol.datatype).__name__}'")
 
-        result = f"{self._nindent}type"
+        result = ""
+        if symbol.preceding_comment != "":
+            for line in symbol.preceding_comment.splitlines():
+                result += f"{self._nindent}{self._COMMENT_PREFIX}{line}\n"
+
+        result += f"{self._nindent}type"
```

Comment:
I added one with a multi-line `preceding_comment`. However I do not test multi-line `inline_comment` as these are (IMHO) wrongly handled by the backend for now, which might need an issue if there isn't one already, actually.

```python
from psyclone.psyir.frontend.fortran import FortranReader
from psyclone.psyir.backend.fortran import FortranWriter

psyir = FortranReader().psyir_from_statement("x = y + z")
psyir.inline_comment = "This is a comment\non two lines"
print(FortranWriter()(psyir))
```
Output:
```
x = y + z  ! This is a comment
on two lines
```

Reply:
Indeed, the parser won't give multiline inline_comments, because it always sets the attribute without appending.
I added an exception in the setter for '\n', to avoid the issue I pointed above in case someone tried to set it that way.
About the last case: only `! ... in the same statement` is kept, and I've added a test for this.